// @mui-treasury/mockup/tabs/UnderlineTabs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UnderlineTabs = void 0;

var _styles = require("@mui/material/styles");

var _Tabs = _interopRequireDefault(require("@mui/material/Tabs"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var UnderlineTabs = (0, _styles.styled)(_Tabs["default"])(function (_ref) {
  var theme = _ref.theme;
  return {
    "&:before": {
      content: '""',
      display: "block",
      height: "1px",
      backgroundColor: theme.palette.grey[200],
      left: 0,
      bottom: 0,
      width: "100%",
      position: "absolute"
    },
    "& .MuiTab-root": {
      textTransform: "none",
      paddingLeft: 0,
      paddingRight: 0,
      marginLeft: theme.spacing(2),
      marginRight: theme.spacing(2),
      "&:first-of-type": {
        marginLeft: 0
      },
      "&:last-of-type": {
        marginRight: 0
      }
    }
  };
});
exports.UnderlineTabs = UnderlineTabs;
