// @mui-treasury/mockup/dashboard/StatCard.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StatCardMockup = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("@mui/material/Box"));

var _shape = require("../shape");

var _typography = require("../typography");

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var StatCardMockup = function StatCardMockup(props) {
  return /*#__PURE__*/_react["default"].createElement(_Box["default"], _extends({}, props, {
    sx: _extends({
      borderRadius: 1,
      border: "1px solid",
      borderColor: _utils.getLightGrey,
      bgcolor: "background.paper",
      p: 1.5,
      display: "flex",
      alignItems: "center",
      boxShadow: "0 2px 6px 0 rgba(0,0,0,0.08)"
    }, props.sx)
  }), /*#__PURE__*/_react["default"].createElement(_shape.Square, null), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    sx: {
      ml: 1.5,
      minWidth: "0px"
    }
  }, /*#__PURE__*/_react["default"].createElement(_typography.Word, {
    sx: {
      height: 10,
      borderRadius: 3,
      bgcolor: _utils.getLightGrey,
      m: 0
    }
  }), /*#__PURE__*/_react["default"].createElement(_typography.Word, {
    sx: {
      mt: 0.5,
      mb: 0
    }
  })));
};

exports.StatCardMockup = StatCardMockup;
