// @mui-treasury/mockup/dashboard/UserActiveCard.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UserActiveCardMockup = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("@mui/material/Box"));

var _LinearProgress = _interopRequireDefault(require("@mui/material/LinearProgress"));

var _Typography = _interopRequireDefault(require("@mui/material/Typography"));

var _shape = require("../shape");

var _typography = require("../typography");

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var UserActiveCardMockup = function UserActiveCardMockup(props) {
  return /*#__PURE__*/_react["default"].createElement(_Box["default"], _extends({}, props, {
    sx: _extends({
      borderRadius: 1,
      border: "1px solid",
      borderColor: _utils.getLightGrey,
      bgcolor: "background.paper",
      p: 1.5,
      boxShadow: "0 2px 6px 0 rgba(0,0,0,0.08)"
    }, props.sx)
  }), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    sx: {
      display: "flex",
      alignItems: "center",
      width: "100%"
    }
  }, /*#__PURE__*/_react["default"].createElement(_typography.Word, {
    sx: {
      bgcolor: _utils.getLightGrey
    }
  }), /*#__PURE__*/_react["default"].createElement(_shape.Circle, {
    sx: {
      fontSize: 20,
      ml: "auto"
    }
  })), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    sx: {
      display: "flex",
      justifyContent: "space-around"
    }
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], null, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    sx: {
      display: "flex",
      alignItems: "center"
    }
  }, /*#__PURE__*/_react["default"].createElement(_shape.Square, {
    sx: {
      fontSize: 16,
      bgcolor: "primary.main",
      mr: 1
    }
  }), /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    variant: "h4"
  }, /*#__PURE__*/_react["default"].createElement("b", null, "92%"))), /*#__PURE__*/_react["default"].createElement(_typography.Word, {
    sx: {
      bgcolor: _utils.getLightGrey,
      ml: 3,
      height: 12,
      borderRadius: 2,
      width: 48
    }
  })), /*#__PURE__*/_react["default"].createElement(_Box["default"], null, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    sx: {
      display: "flex",
      alignItems: "center"
    }
  }, /*#__PURE__*/_react["default"].createElement(_shape.Square, {
    sx: {
      fontSize: 16,
      bgcolor: "error.main",
      mr: 1
    }
  }), /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    variant: "h4"
  }, /*#__PURE__*/_react["default"].createElement("b", null, "8%"))), /*#__PURE__*/_react["default"].createElement(_typography.Word, {
    sx: {
      bgcolor: _utils.getLightGrey,
      ml: 3,
      height: 12,
      borderRadius: 2,
      width: 48
    }
  }))), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_LinearProgress["default"], {
    variant: "determinate",
    value: 92,
    sx: {
      height: 10,
      borderRadius: 3,
      bgcolor: _utils.getLightGrey,
      "& > .MuiLinearProgress-bar": {
        borderRadius: 3
      }
    }
  }), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    height: 16
  }), /*#__PURE__*/_react["default"].createElement(_LinearProgress["default"], {
    variant: "determinate",
    value: 8,
    sx: {
      height: 10,
      borderRadius: 3,
      bgcolor: _utils.getLightGrey,
      "& > .MuiLinearProgress-bar": {
        borderRadius: 3,
        bgcolor: "error.main"
      }
    }
  }), /*#__PURE__*/_react["default"].createElement("br", null));
};

exports.UserActiveCardMockup = UserActiveCardMockup;
