// @mui-treasury/mockup/layout/FooterMockup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FooterMockup = exports.LinkGroup = void 0;

var _react = _interopRequireDefault(require("react"));

var _Container = _interopRequireDefault(require("@mui/material/Container"));

var _Box = _interopRequireDefault(require("@mui/material/Box"));

var _Grid = _interopRequireDefault(require("@mui/material/Grid"));

var _Stack = _interopRequireDefault(require("@mui/material/Stack"));

var _shape = require("../shape");

var _utils = require("../utils");

var _excluded = ["disableGutters"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && Symbol.iterator in Object(iter)) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var LinkGroup = function LinkGroup(_ref) {
  var _ref$lineCount = _ref.lineCount,
      lineCount = _ref$lineCount === void 0 ? 5 : _ref$lineCount,
      sx = _ref.sx;
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_shape.Pill, {
    sx: {
      height: 12,
      width: 40
    }
  }), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Stack["default"], {
    spacing: 2
  }, _toConsumableArray(Array(lineCount)).map(function (_, index) {
    return /*#__PURE__*/_react["default"].createElement(_shape.Square, {
      key: index,
      sx: {
        width: (0, _utils.randomBetween)(56, 120),
        height: 16
      }
    });
  })));
};

exports.LinkGroup = LinkGroup;

var FooterMockup = function FooterMockup(_ref2) {
  var _ref2$disableGutters = _ref2.disableGutters,
      disableGutters = _ref2$disableGutters === void 0 ? false : _ref2$disableGutters,
      props = _objectWithoutProperties(_ref2, _excluded);

  return /*#__PURE__*/_react["default"].createElement(_Box["default"], _extends({}, props, {
    sx: _extends({
      borderTop: "1px solid",
      borderColor: function borderColor(theme) {
        return theme.palette.mode === "dark" ? "rgba(255,255,255,0.12)" : "grey.200";
      },
      bgcolor: function bgcolor(theme) {
        return theme.palette.mode === "dark" ? "rgba(255,255,255,0.07)" : "grey.50";
      }
    }, props.sx)
  }), /*#__PURE__*/_react["default"].createElement(_Container["default"], {
    sx: {
      py: 5,
      minHeight: 300
    },
    disableGutters: disableGutters
  }, /*#__PURE__*/_react["default"].createElement(_Grid["default"], {
    container: true,
    spacing: 2
  }, /*#__PURE__*/_react["default"].createElement(_Grid["default"], {
    item: true,
    xs: 12,
    sm: 6,
    md: 3,
    sx: {
      order: {
        xs: 2,
        sm: 1,
        md: 1
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(_shape.Square, {
    sx: {
      width: 72,
      height: 48
    }
  }), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_shape.Pill, {
    sx: {
      height: 14
    }
  }), /*#__PURE__*/_react["default"].createElement("br", null)), /*#__PURE__*/_react["default"].createElement(_Grid["default"], {
    item: true,
    xs: 6,
    md: 3,
    sx: {
      order: {
        xs: 3,
        md: 2
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(LinkGroup, null)), /*#__PURE__*/_react["default"].createElement(_Grid["default"], {
    item: true,
    xs: 6,
    md: 3,
    sx: {
      order: {
        xs: 4,
        md: 3
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(LinkGroup, null)), /*#__PURE__*/_react["default"].createElement(_Grid["default"], {
    item: true,
    xs: 12,
    sm: 6,
    md: 3,
    sx: {
      order: {
        xs: 1,
        sm: 2,
        md: 4
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(_Stack["default"], {
    direction: "row",
    spacing: 1,
    sx: {
      justifyContent: {
        xs: "center",
        sm: "flex-start"
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(_shape.Square, {
    sx: {
      height: 36,
      flexBasis: 160,
      flexShrink: 1
    }
  }), /*#__PURE__*/_react["default"].createElement(_shape.Square, {
    sx: {
      height: 36,
      flexBasis: 56,
      flexShrink: 1
    }
  })), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Stack["default"], {
    direction: "row",
    spacing: 1,
    sx: {
      justifyContent: {
        xs: "center",
        sm: "flex-start"
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(_shape.Circle, null), /*#__PURE__*/_react["default"].createElement(_shape.Circle, null), /*#__PURE__*/_react["default"].createElement(_shape.Circle, null)), /*#__PURE__*/_react["default"].createElement("br", null)))));
};

exports.FooterMockup = FooterMockup;
