// @mui-treasury/mockup/layout/HeaderMockup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HeaderMockup = void 0;

var _react = _interopRequireDefault(require("react"));

var _Avatar = _interopRequireDefault(require("@mui/material/Avatar"));

var _Box = _interopRequireDefault(require("@mui/material/Box"));

var _shape = require("../shape");

var _excluded = ["trigger"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var HeaderMockup = function HeaderMockup(_ref) {
  var trigger = _ref.trigger,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_react["default"].createElement(_Box["default"], _extends({}, props, {
    sx: _extends({
      display: "flex",
      alignItems: "center",
      justifyContent: {
        xs: "space-between",
        sm: "flex-start"
      },
      gap: 2,
      px: 2,
      flex: 1
    }, props.sx)
  }), trigger, /*#__PURE__*/_react["default"].createElement(_shape.Square, null), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    ml: "auto",
    sx: {
      display: {
        xs: "none",
        sm: "flex"
      },
      alignItems: "center",
      gap: {
        xs: 2,
        md: 3
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(_shape.Pill, null), /*#__PURE__*/_react["default"].createElement(_shape.Pill, null), /*#__PURE__*/_react["default"].createElement(_shape.Pill, null), /*#__PURE__*/_react["default"].createElement(_shape.Pill, null)), /*#__PURE__*/_react["default"].createElement(_Avatar["default"], {
    sx: {
      width: 32,
      height: 32,
      ml: {
        xs: 0,
        sm: 2
      }
    }
  }));
};

exports.HeaderMockup = HeaderMockup;
